src/storage/src/storage_state.rs

```diff
@@ -4,12 +4,67 @@
 // included in the LICENSE file.
 
 //! Worker-local state for storage timely instances.
+//!
+//! One instance of a [`Worker`], along with its contained [`StorageState`], is
+//! part of an ensemble of storage workers that all run inside the same timely
+//! cluster. We call this worker a _storage worker_ to disambiguate it from
+//! other kinds of workers, potentially other components that might be sharing
+//! the same timely cluster.
+//!
+//! ## Controller and internal communication
+//!
+//! A worker receives _external_ [`StorageCommands`](StorageCommand) from the
```

Comment:
No, all workers receive external commands. I implemented something similar to what Petros suggested here: https://github.com/MaterializeInc/materialize/pull/16958#discussion_r1064014631.

This is where we make sure that we only report once we have received the _external_ command on a given worker: https://github.com/MaterializeInc/materialize/pull/16958/files?diff=split&w=0#diff-752224caae5760da0459a31cacb0fe5853e6c17de0b4027598e8da60f9c5f092R583

cc @bkirwi because this is relevant for a question you asked elsewhere